test/unit/auth/auth.spec.ts

```diff
@@ -328,63 +328,49 @@ AUTH_CONFIGS.forEach((testConfig) => {
     }
 
     describe('createCustomToken()', () => {
-      let spy: sinon.SinonSpy;
-      beforeEach(() => {
-        spy = sinon.spy(FirebaseTokenGenerator.prototype, 'createCustomToken');
-      });
-
-      afterEach(() => {
-        spy.restore();
+      it('should return a jwt', async () => {
+        const token = await auth.createCustomToken('uid1');
+        const decodedToken = jwt.decode(token, {complete: true});
+        expect(decodedToken).to.have.property('header').that.has.property('typ', 'JWT');
       });
 
       if (testConfig.Auth === TenantAwareAuth) {
-        it('should reject with an unsupported tenant operation error', () => {
-          const expectedError = new FirebaseAuthError(AuthClientErrorCode.UNSUPPORTED_TENANT_OPERATION);
-          return auth.createCustomToken(mocks.uid)
-            .then(() => {
-              throw new Error('Unexpected success');
-            })
-            .catch((error) => {
-              expect(error).to.deep.equal(expectedError);
-            });
+        it('should contain tenant_id', async () => {
+          const token = await auth.createCustomToken('uid1');
+          expect(jwt.decode(token)).to.have.property('tenant_id', TENANT_ID);
         });
       } else {
-        it('should throw if a cert credential is not specified', () => {
-          const mockCredentialAuth = testConfig.init(mocks.mockCredentialApp());
-
-          expect(() => {
-            mockCredentialAuth.createCustomToken(mocks.uid, mocks.developerClaims);
-          }).not.to.throw;
+        it('should not contain tenant_id', async () => {
+          const token = await auth.createCustomToken('uid1');
+          expect(jwt.decode(token)).to.not.have.property('tenant_id');
         });
+      }
 
-        it('should forward on the call to the token generator\'s createCustomToken() method', () => {
-          const developerClaimsCopy = deepCopy(mocks.developerClaims);
-          return auth.createCustomToken(mocks.uid, mocks.developerClaims)
-            .then(() => {
-              expect(spy)
-                .to.have.been.calledOnce
-                .and.calledWith(mocks.uid, developerClaimsCopy);
-            });
-        });
+      it('should throw if a cert credential is not specified', () => {
+        const mockCredentialAuth = testConfig.init(mocks.mockCredentialApp());
 
-        it('should be fulfilled given an app which returns null access tokens', () => {
-          // createCustomToken() does not rely on an access token and therefore works in this scenario.
-          return nullAccessTokenAuth.createCustomToken(mocks.uid, mocks.developerClaims)
-            .should.eventually.be.fulfilled;
-        });
+        expect(() => {
+          mockCredentialAuth.createCustomToken(mocks.uid, mocks.developerClaims);
+        }).not.to.throw;
```

Comment:
Seems to be a mismatch with the test description (should throw if...)

Reply:
But wait! That's wrong too, since this test can't call "the IAM service in the cloud when the service account credentials are not provided." So instead of throwing immediately, it now returns a failed future. Fixed (for real this time) by ensuring it *does* fail, but not immediately. (Might be worth checking with @hiranya911 to see if this captures the original intent.)